### ISSUE TITLE
htb class cannot limit rate

### ISSUE THREAD

**Issue by yandd** [vishvananda/netlink#258]
demo is here, https://play.golang.org/p/sCA-isfWEw
```go
		class := netlink.NewHtbClass(netlink.ClassAttrs{
			LinkIndex: link.Attrs().Index,
			Handle:    classidHandle,
			Parent:    parentHandle,
		}, netlink.HtbClassAttrs{
			Rate: uint64(*rate),
			Ceil: uint64(*ceil),
		})
		err = netlink.ClassReplace(class)
		if err != nil {
			fmt.Println("del class failed:", err)
			os.Exit(1)
		}
```
```
./tc_class -replace -dev=eth0 -parent=1: -classid=1:1 -rate=1000000 -ceil=1000000
```
The command above cannot limit the rate, but use 'tc' command can query the class info, and 'tc' command can limit the rate
```
tc -s -d class show dev eth0
```

test case：
```
tc qdisc add dev eth0 root handle 1: htb r2q 1
tc class replace dev eth0 parent 1: classid 1:1 htb rate 1mbit ceil 1mbit burst 1mbit
tc filter add dev eth0 parent 1: prio 1 protocol ip u32 match ip src $LIMIT_IP flowid 1:1
```

**Comment by yandd** [comment]
I get kernel info here:
```
Aug 21 12:03:02 10-9-197-37 kernel: [2058226.757937] netlink: 12 bytes leftover after parsing attributes.
Aug 21 12:03:02 10-9-197-37 kernel: [2058226.757940] netlink: 12 bytes leftover after parsing attributes.
Aug 21 12:03:02 10-9-197-37 kernel: [2058226.757961] netlink: 12 bytes leftover after parsing attributes.
Aug 21 12:03:02 10-9-197-37 kernel: [2058226.757976] netlink: 12 bytes leftover after parsing attributes.
Aug 21 12:03:02 10-9-197-37 kernel: [2058226.757984] netlink: 12 bytes leftover after parsing attributes.
```

my os is `CentOS release 6.5 (Final)`:
Linux 10-9-145-128 2.6.32-431.11.25.el6.ucloud.x86_64 #1 SMP Tue Jul 19 10:06:12 EDT 2016 x86_64 x86_64 x86_64 GNU/Linux

**Comment by yandd** [comment]
in `filter_class.go`, you cannot do that:
```go
func CalcRtable(rate *nl.TcRateSpec, rtab [256]uint32, cellLog int, mtu uint32, linklayer int) int {
	bps := rate.Rate
	mpu := rate.Mpu
	var sz uint
	if mtu == 0 {
		mtu = 2047
	}
	if cellLog < 0 {
		cellLog = 0
		for (mtu >> uint(cellLog)) > 255 {
			cellLog++
		}
	}
	for i := 0; i < 256; i++ {
		sz = AdjustSize(uint((i+1)<<uint32(cellLog)), uint(mpu), linklayer)
		rtab[i] = uint32(Xmittime(uint64(bps), uint32(sz)))
	}
	rate.CellAlign = -1
	rate.CellLog = uint8(cellLog)
	rate.Linklayer = uint8(linklayer & nl.TC_LINKLAYER_MASK)
	return cellLog
}
```

It must be like this:
```go
func CalcRtable(rate *nl.TcRateSpec, rtab *[256]uint32, cellLog int, mtu uint32, linklayer int) int {
	bps := rate.Rate
	mpu := rate.Mpu
	var sz uint
	if mtu == 0 {
		mtu = 2047
	}
	if cellLog < 0 {
		cellLog = 0
		for (mtu >> uint(cellLog)) > 255 {
			cellLog++
		}
	}
	for i := 0; i < 256; i++ {
		sz = AdjustSize(uint((i+1)<<uint32(cellLog)), uint(mpu), linklayer)
		(*rtab)[i] = uint32(Xmittime(uint64(bps), uint32(sz)))
	}
	rate.CellAlign = -1
	rate.CellLog = uint8(cellLog)
	rate.Linklayer = uint8(linklayer & nl.TC_LINKLAYER_MASK)
	return cellLog
}
```

**Comment by yandd** [comment]
I fixed it in pr #259 , but `class replace` still cannot work

**Comment by yandd** [comment]
Solved with commit aa48b8cff04060fa0c2e4eea0e87c71d3d5b0876
